react-front/public/components/ConfigChange/ConfigChangeStep2.js

```diff
@@ -0,0 +1,145 @@
+    repeatingData: [] // setInterval ID saved in state to be accessible to all class methods
+    // errorMessage: ""
+  };
+
+  deviceSyncTo = () => {
+    const credentials = this.state.token;
+    console.log("you clicked the start sync info button");
+    let url = "https://tug-lab.cnaas.sunet.se:8443/api/v1.0/device_syncto";
+    let dataToSend = { dry_run: true, all: true };
+    postData(url, credentials, dataToSend).then(data => {
+      console.log("this should be data", data);
+      {
+        this.setState(
+          {
+            syncData: data
+          },
+          () => {
+            this.syncStatus();
+          },
+          () => {
+            console.log("this is new state", this.state.syncData);
+          }
+        );
+      }
+    });
+  };
+
+  syncStatus = () => {
+    // use jobId from API in url
+    // let jobId = this.state.syncData.job_id;
+    // mock jobId that returns relevant data
+    let jobId = "5ddbe1548b2d390c963b97d8";
+    const credentials = this.state.token;
+    console.log("this API call is automatic");
+    let url = `https://tug-lab.cnaas.sunet.se:8443/api/v1.0/job/${jobId}`;
+
+    let repeatingData = setInterval(() => {
+      getData(url, credentials).then(data => {
+        console.log("this should be data.data.jobs", data.data.jobs);
+        {
+          this.setState(
+            {
+              jobsData: data.data.jobs,
+              repeatingData: repeatingData
+            },
+            () => {
+              console.log("this is jobs data", this.state.jobsData);
+            }
+          );
+        }
+      });
+    }, 500);
+  };
+
+  randomIntFromInterval(min, max) {
+    return Math.floor(Math.random() * (max - min + 1) + min);
+  }
+
+  render() {
+    // extract sync data
+    let syncData = this.state.syncData;
+    let syncMessage = syncData.data;
+    let syncStatus = syncData.status;
+    let syncJobId = syncData.job_id;
+
+    // iterate trhough the job data and extract relevant info
+    let jobStatus = "";
+    let jobStartTime = "";
+    let jobFinishTime = "";
+    // totalDevices mocked at 100 for progress bar
+    let finishedDevices = 0;
+    let totalDevices = 100;
+
+    let jobsData = this.state.jobsData;
+    jobsData.map((job, i) => {
+      jobStatus = job.status;
+      jobStartTime = job.start_time;
+      jobFinishTime = job.finish_time;
+
+      // generating random data to mock value in progress bar
+      finishedDevices = this.randomIntFromInterval(0, 100);
+      console.log("this is finsihedDevices", finishedDevices);
+      // extracting actual data when API populated
+      // let finishedDevices = job.finished_devices.length;
+      // let totalDevices = job.result._totals.selected_devices;
+      // let exceptionText = job.exception;
+
+      // stop the setInterval when job status is finished
+      if (jobStatus === "FINISHED") {
+        console.log("jobStatus is finished");
+        clearInterval(this.state.repeatingData);
+      }
```

Comment:
If there's an error the job will exit with status EXCEPTION and never reach status FINISHED, so it's probably a good idea to check for that as well so this doesn't run forever

Reply:
ah ok! I'll add that to the halt...